### PR TITLE
Support for Astro language server

### PR DESCRIPTION
**Pull request by pawelkobojek**
Support for [Astro](https://astro.build/) [language server](https://github.com/withastro/language-tools). Related issue: #3584

### CODE REVIEW

**Comment by lucafanselau** [comment]
Just as a note, to enable typescript support in the astro-language-server you ll have to pass in initialization options. You can do something along the lines of:

`(defun astro-get-tsserver ()
    ""
    (f-join (lsp-workspace-root) "node_modules/typescript/lib/tsserverlibrary.js"))

  (lsp-register-client
   (make-lsp-client :new-connection (lsp-stdio-connection '("astro-ls" "--stdio"))
                    :activation-fn (lsp-activate-on "astro")
                    :initialization-options (lambda ()
                                              `(:typescript (:serverPath ,(astro-get-tsserver))))
                    :server-id 'astro-ls))`

initalization options are inspired by the astro plugin

File: clients/lsp-astro.el [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+;;; Code:
+(require 'lsp-mode)
+
+(defgroup lsp-astro nil
+  "LSP support for Astro.build, using astro-ls."
+  :group 'lsp-mode
+  :link '(url-link "https://github.com/withastro/language-tools"))
+
+(add-to-list 'lsp-language-id-configuration
```
Comment (yyoncho) [inline clients/lsp-astro.el:33]: can you move this in lsp-language-id-configuration initialization?

Comment (pawelkobojek) [reply clients/lsp-astro.el]: Done

**Comment by pawelkobojek** [comment]
@lucafanselau Sure, go ahead, I'd appreciate your help.

**Comment by lucafanselau** [comment]
> @lucafanselau Sure, go ahead, I'd appreciate your help.

Okay can do. I would need either push access to your fork, or I would have to open another pr. Let me know which works for you!

**Comment by pawelkobojek** [comment]
Just sent you an invite

**Comment by yyoncho** [comment]
If you want, I can merge it and then you can open a new PR.

**Comment by pawelkobojek** [comment]
@yyoncho I agree, you can merge it

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
@pawelkobojek, thank you for contributing to `lsp-mode`, and congrats on your first contribution.

**Pull request merged**